tests/test_zonemap.py

```diff
@@ -45,74 +46,123 @@ def test_nonexistingzones():
+    assert "Could not parse lyr file" in caplog.text
     assert "Failed on content: foo" in caplog.text
 
-    zonefile = tmpdir / "zonemap.lyr"
-    zonefile.write_text(
+    lyrfile = tmpdir / "formations.lyr"
+    lyrfile.write_text(
         """
 valid 1-2
 foo 1 2 3
 """,
         encoding="utf-8",
     )
-    assert ecl2df.common.parse_zonemapfile(zonefile) is None
+    assert ecl2df.common.parse_lyrfile(lyrfile) is None
     assert "Failed on content: foo 1 2 3" in caplog.text
 
-    zonefile = tmpdir / "zonemap.lyr"
-    zonefile.write_text(
+    lyrfile = tmpdir / "formations.lyr"
+    lyrfile.write_text(
         """
-valid 1-2 stray
+foo 2-1
 """,
         encoding="utf-8",
     )
-    assert ecl2df.EclFiles(DATAFILE).get_zonemap(str(zonefile)) is None
-    assert "Failed on content: valid 1-2 stray" in caplog.text
+    assert ecl2df.EclFiles(DATAFILE).get_zonemap(str(lyrfile)) is None
+    assert "From_layer higher than to_layer" in caplog.text
+
+    lyrfile = tmpdir / "formations.lyr"
+    lyrfile.write_text(
+        """
+valid 1-2 #FFE5F7
+foo   3- 4 #FFGGHH
+""",
+        encoding="utf-8",
+    )
+    assert ecl2df.EclFiles(DATAFILE).get_zonemap(str(lyrfile)) is None
+    assert "Failed on content: foo   3- 4 #FFGGHH" in caplog.text
 
 
-def test_spaces_dashes(tmpdir):
-    """Ensure we support dashes around dashes"""
-    zonefile = tmpdir / "zonemap.lyr"
-    zonefile.write_text(
+def test_lyrlist_format(tmpdir):
+    """Ensure the lyr file is parsed correctly"""
+    lyrfile = tmpdir / "formations.lyr"
+    lyrfile.write_text(
+        """
+-- Some text
+'ZoneA'          1 -     5  #FFE5F7
+'ZoneB'          6-     10  --no color
+'ZoneC'          11-15    blue
+'ZoneD'         3          #fbb
+'ZoneE'         19     -20
+'ZoneF'         21-22  CORNFLOWERBLUE
+""",
+        encoding="utf-8",
+    )
+    lyrlist = ecl2df.common.parse_lyrfile(lyrfile)
+    # 0 - ZoneA
+    assert lyrlist[0]["from_layer"] == 1
+    assert lyrlist[0]["color"] == "#FFE5F7"
+    assert lyrlist[0]["name"] == "ZoneA"
+    # 1 - ZoneB
+    assert lyrlist[1]["to_layer"] == 10
+    assert "color" not in lyrlist[1]
+    # 2 - ZoneC
+    assert lyrlist[2]["color"] == "blue"
+    # 3 - ZoneD
+    assert lyrlist[3]["color"] == "#fbb"
+    assert lyrlist[3]["span"] == 3
+    assert "from_layer" not in lyrlist[3]
+    assert "to_layer" not in lyrlist[3]
+    # 4 - ZoneE
+    assert lyrlist[4]["to_layer"] == 20
+    assert "color" not in lyrlist[4]
+    # 5 - ZoneF
+    assert lyrlist[5]["color"] == "CORNFLOWERBLUE"
+
+
+def test_convert_lyrlist_to_zonemap(tmpdir):
+    """Test common.covert_lyrlist_to_zonemap()"""
+    lyrfile = tmpdir / "formations.lyr"
+    lyrfile.write_text(
         """
--- Layer table for 83 layer simulation model xxxx
-'UT3_3'          1 -     15
-'UT3_2'          16 -   20
-'UT3_1'         21 -    21
-'UT2'           22 -   23
-'UT1_2'         24 -   37
-'UT1_1'         38 -   46 -- thistextisignored
```

Comment:
looks like this corner case has been lost in the edited version? (comment-at-end-of-line)

Reply:
Looks like it is covered further up, in ZoneB

Reply:
I thought it was coverd by the ZoneB line here? and the test `assert "color" not in lyrlist[1]`

```
def test_lyrlist_format(tmpdir):
    """Ensure the lyr file is parsed correctly"""
    lyrfile = tmpdir / "formations.lyr"
    lyrfile.write_text(
        """
-- Some text
'ZoneA'          1 -     5  #FFE5F7
'ZoneB'          6-     10  --no color
'ZoneC'          11-15    blue
'ZoneD'         3          #fbb
'ZoneE'         19     -20
'ZoneF'         21-22  CORNFLOWERBLUE
""",
        encoding="utf-8",
    )
    lyrlist = ecl2df.common.parse_lyrfile(lyrfile)
    # 0 - ZoneA
    assert lyrlist[0]["from_layer"] == 1
    assert lyrlist[0]["color"] == "#FFE5F7"
    assert lyrlist[0]["name"] == "ZoneA"
    # 1 - ZoneB
    assert lyrlist[1]["to_layer"] == 10
    assert "color" not in lyrlist[1]
```